build_extra/rust/BUILD.gn

```diff
@@ -8,1145 +8,1786 @@
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/x86_64-mont5-macosx.S",
+    ]
+  }
+  if (is_win) {
+    libs = [
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/aes-x86_64-nasm.obj",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/aesni-gcm-x86_64-nasm.obj",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/aesni-x86_64-nasm.obj",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/chacha-x86_64-nasm.obj",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/ghash-x86_64-nasm.obj",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/p256-x86_64-asm-nasm.obj",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/poly1305-x86_64-nasm.obj",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/sha256-x86_64-nasm.obj",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/sha512-x86_64-nasm.obj",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/vpaes-x86_64-nasm.obj",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/x86_64-mont-nasm.obj",
+      "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/pregenerated/x86_64-mont5-nasm.obj",
+    ]
+
+    # Suppress "warning: '_addcarry_u64' is not a recognized builtin."
+    cflags = [ "-Wno-ignored-pragma-intrinsic" ]
+  }
+}
+
+## Override: don't build 'ring-test' static library.
+# static_library("ring-test") {
+#   include_dirs = [ "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/include" ]
+#   sources = [ "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/crypto/constant_time_test.c" ]
+# }
+
+rust_crate("rustls") {
+  source_root = "$cargo_home/registry/src/github.com-1ecc6299db9ec823/rustls-0.14.0/src/lib.rs"
+  features = [
+    "default",
+    "log",
+    "logging",
+  ]
+  extern = [
+    ":base64",
     ":log",
-    ":net2",
-    ":time",
-    ":tokio",
-    ":tokio_executor",
-    ":tokio_io",
-    ":tokio_reactor",
-    ":tokio_tcp",
-    ":tokio_threadpool",
-    ":tokio_timer",
-    ":want",
+    ":ring",
+    ":sct",
+    ":untrusted",
+    ":webpki",
+  ]
+  args = [
+    "--cap-lints",
+    "allow",
   ]
 }
 
-rust_crate("h2") {
-  source_root = "$registry_github/h2-0.1.14/src/lib.rs"
+rust_crate("rustyline") {
+  source_root = "$cargo_home/registry/src/github.com-1ecc6299db9ec823/rustyline-2.1.0/src/lib.rs"
   extern = [
-    ":byteorder",
-    ":bytes",
-    ":fnv",
-    ":futures",
-    ":http",
-    ":indexmap",
+    ":dirs",
+    ":libc",
     ":log",
-    ":slab",
-    ":string",
-    ":tokio_io",
+    ":memchr",
+    ":unicode_segmentation",
+    ":unicode_width",
+  ]
+  args = [
+    "--cap-lints",
+    "allow",
```

Comment:
What's "--cap-lints allow" do?

Reply:
Surpresses lints (warnings etc). Cargo uses this to ensure that lints introduced in a newer rust version don't break crates that were written with an older rust version in mind. 

Before, we would add '-Asomething' manually to achieve the same.